Error while checking build/O2/fullCI for 5ee4098581dbac30c31d9a76e9dd38861684053f at 2024-01-15 23:45:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2-full-system-test-latest/log
command /sw/slc8_x86-64/O2/12499-slc8_x86-64-local1/prodtests/full-system-test/dpl-workflow.sh had nonzero exit code 128
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data HMP/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data FLP/DISTSUBTIMEFRAME/0 was not created for timeslice 0 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data MID/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data CTP/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data TOF/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data TPC/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data ITS/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data MFT/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data MCH/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data PHS/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data TRD/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data CPV/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data EMC/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data FDD/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data FT0/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data FV0/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data ZDC/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data FLP/DISTSUBTIMEFRAME/52443 was not created for timeslice 1 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data HMP/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data FLP/DISTSUBTIMEFRAME/0 was not created for timeslice 1 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data MID/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data CTP/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data TOF/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data TPC/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data ITS/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data MFT/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data MCH/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data PHS/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data TRD/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data CPV/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data EMC/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data FDD/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data FT0/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data FV0/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[49295:raw-file-reader]: [22:44:35][ERROR] Expected Lifetime::Timeframe data ZDC/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[49324:BadMapCalibSpec]: [22:45:04][ERROR] Insufficient statistics: 4 entries in lowE histo, do nothing
[49387:ctf-writer]: [22:45:06][ERROR] Some Lifetime::Timeframe data got dropped starting at 1
[ERROR] Workflow crashed - PID 49376 (tpc-calibrator-gainmap-tracks) did not exit correctly however it's not clear why. Exit code forced to 128.
[ERROR]  - Device tpc-calibrator-gainmap-tracks: pid 49376 (exit 128)
[ERROR] SEVERE: Device tpc-calibrator-gainmap-tracks (49376) returned with 128
[0 more errors; see full log]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12499/5ee4098581dbac30c31d9a76e9dd38861684053f/build_O2_fullCI/pretty.html).

--- a/Framework/Core/src/FairMQDeviceProxy.cxx
+++ b/Framework/Core/src/FairMQDeviceProxy.cxx
@@ -320,10 +320,7 @@ void FairMQDeviceProxy::bind(std::vector<OutputRoute> const& outputs, std::vecto
       mInputRoutes.emplace_back(RouteState{channelIndex, false});
       ri++;
     }
-    for (auto& route : mInputRoutes) {
-      assert(route.channel.value != -1);
-      assert(route.channel.value < mInputChannels.size());
-    }
+    assert(std::all_of(mInputRoutes.begin(), mInputRoutes.end(), [s = mInputChannels.size()](RouteState const& route) { return route.channel.value != -1 && route.channel.value < s; }));
     LOGP(detail, "Total input channels found {}, total routes {}", mInputChannels.size(), mInputRoutes.size());
     assert(mInputRoutes.size() == inputs.size());
   }

